crossbar/router/broker.py

```diff
@@ -224,6 +231,25 @@ def on_authorize_success(authorization):
                     if store_event:
                         self._event_store.store_event(session._session_id, publication, publish.topic, publish.args, publish.kwargs)
 
+                    # retain event on the topic
+                    #
+                    if retain_event:
+                        observation = self._subscription_map.get_observation(publish.topic, create_if_non_existing=True)
+
+                        if not observation.extra:
+                            observation.extra = {}
```

Comment:
no, pls not a dict, but a class .. similar to how it is done here: https://github.com/crossbario/crossbar/blob/master/crossbar/router/dealer.py#L72 and the lines down inside the dealer ..